src/lib.rs

```diff
@@ -185,52 +189,163 @@ fn enforce_container_security_policies(
+                                serde_json::from_value::<ReplicationController>(
+                                    validation_request.request.object.clone(),
+                                )?;
+                            if let Some(ref mut spec) = replication_controller.spec {
+                                if let Some(template) = &spec.template {
+                                    let mut new_template = template.clone();
+                                    new_template.spec = Some(pod_spec);
+                                    spec.template = Some(new_template)
+                                }
+                            }
+                            return kubewarden::mutate_request(serde_json::to_value(
+                                replication_controller,
+                            )?);
+                        }
+                        CronJob::KIND => {
+                            let mut cronjob = serde_json::from_value::<CronJob>(
+                                validation_request.request.object.clone(),
+                            )?;
+                            if let Some(ref mut spec) = cronjob.spec {
+                                let mut new_job_template = spec.job_template.clone();
+                                if let Some(ref mut spec) = new_job_template.spec {
+                                    spec.template.spec = Some(pod_spec);
+                                }
+                                spec.job_template = new_job_template
+                            }
+                            return kubewarden::mutate_request(serde_json::to_value(cronjob)?);
+                        }
+                        Job::KIND => {
+                            let mut job = serde_json::from_value::<Job>(
+                                validation_request.request.object.clone(),
+                            )?;
+                            if let Some(ref mut spec) = job.spec {
+                                spec.template.spec = Some(pod_spec);
+                            }
+                            return kubewarden::mutate_request(serde_json::to_value(job)?);
+                        }
+                        Pod::KIND => {
+                            let mut pod = serde_json::from_value::<Pod>(
+                                validation_request.request.object.clone(),
+                            )?;
+                            pod.spec = Some(pod_spec);
+                            return kubewarden::mutate_request(serde_json::to_value(pod)?);
+                        }
+                        _ => {
+                            return kubewarden::reject_request(Some("Object should be one of these kinds: Deployment, ReplicaSet, StatefulSet, DaemonSet, ReplicationController, Job, CronJob, Pod".to_string()), None, None, None);
+                        }
+                    };
+                } else {
+                    return kubewarden::accept_request();
```

Comment:
Should we move the code to mutate each resource type into the SKD?

Reply:
Yes, I think it would be nice to do that